src/main/java/com/inin/analytics/elasticsearch/ESEmbededContainer.java

```diff
@@ -71,12 +72,49 @@ public void snapshot(List<String> indicies, String snapshotName, String snapshot
 
 		// Snapshot
 		long start = System.currentTimeMillis();
-		node.client().admin().cluster().prepareCreateSnapshot(snapshotRepoName, snapshotName).setWaitForCompletion(true).setIndices((String[]) indicies.toArray(new String[0])).get();
+		node.client().admin().cluster().prepareCreateSnapshot(snapshotRepoName, snapshotName).setIndices((String[]) indicies.toArray(new String[0])).execute();
+
+		// ES snapshot restore ignores timers and will block no more than 30s :( You have to block & poll to make sure it's done
+		blockForSnapshot(snapshotRepoName, indicies, timeoutMS, reporter);	
+		
 		if(reporter != null) {
 			reporter.incrCounter(BaseESReducer.JOB_COUNTER.TIME_SPENT_SNAPSHOTTING_MS, System.currentTimeMillis() - start);
 		}
 
 	}
+
+	/** 
+	 * Block for index snapshots to be complete
+	 *  
+	 * @param snapshotRepoName
+	 * @param index
+	 * @param timeoutMS
+	 * @param reporter
+	 */
+	private void blockForSnapshot(String snapshotRepoName, List<String> indicies, long timeoutMS, Reporter reporter) {
+		long start = System.currentTimeMillis();
+		while(System.currentTimeMillis() - start < timeoutMS) {
+
+			GetSnapshotsResponse repos = node.client().admin().cluster().getSnapshots(new GetSnapshotsRequest(snapshotRepoName)).actionGet();
+				for(SnapshotInfo i : repos.getSnapshots()) {
+					if(i.state().completed() && i.successfulShards() == i.totalShards() && i.totalShards() >= indicies.size()) {
+						logger.info("Snapshot completed {} out of {} indicies. Snapshot state {}. ", i.successfulShards(), i.totalShards(), i.state().completed());
+						return;
+					} else {
+						logger.info("Snapshotted {} out of {} indicies, polling for completion. Snapshot state {}.", i.successfulShards(), i.totalShards(), i.state().completed());
+					}
+				}
+			try {
+				if(reporter != null) {
+					reporter.incrCounter(BaseESReducer.JOB_COUNTER.TIMES_BLOCKED_WAITING_FOR_SNAPSHOT, 1);
```

Comment:
Suggestion.  Maybe the time spent waiting would be a better metric?  Right now, the amount of time I'm sleeping each cycle dictates what this counter really means.